Objective4_WAL3: complete Milestone 1 with full CRUD, CSV logic, and Jest test.

arunkjoyson opened the pull request:
- Implemented add, edit, and delete features for evaluations
- Integrated CSV file handling for data persistence
- Added user feedback via ConfirmationMessage component
- Created test cases for form, table, and message components using Jest
- Updated README with full structure, instructions, and test coverage

achow-conestogac reviewed (commented):
In general, I do not like the whole "component", "tests" directory organization.

this tells me nothing about the application except that components and tests exist.

Group your tests and implementation like so:

```
| src
| -- Components
|---- ConfirmationMessage
|------ConfirmationMessage.tsx
|------ConfirmationMessage.test.tsx
```

The next thing is the difference between confirmation message component vs EvaluationForms and Evaluations tables.

components are designed to be reusable components... which `ConfirmationMessage.tsx` is. 

`EvaluationForm.tsx` on the other hand is a lot more static and serves the purpose of YOUR specific workflow.  As such, it should not belong in the `components` directory.

Additionally, this goes for `types` as well.  group your types into files where it is used.... I would argue that it should not be a separate file at all but something to be included in either the `EvaluationForm.tsx` or the `EvaluationTable.tsx`.

Last note here is that all the comments talking about for which issue something is coded for needs to be removed.  Comments are there to tell you WHY you did something... not WHAT you did.  It should also be used ONLY when you absolutely CANNOT express this by creating better programming flows.  

On the plus side, your tests look decent. 

Please do this and I can review again.

arunkjoyson commented:
Hi Andy,

Thank you for the feedback! we have refactored the code accordingly.

- Moved ConfirmationMessage, EvaluationForm, EvaluationTable into feature-based folders
- Consolidated types into respective component files
- Renamed utils directory to csvHandler for clarity
- Moved validation logic into EvaluationFormValidation.ts
- Updated App.tsx to only include EvaluationScreen
- Removed redundant comments and improved code readability
- Removed unwanted README.md content and added the original one

thanks!

arunkjoyson commented on src/utils/csvHandler.ts:
done

arunkjoyson commented on src/components/EvaluationTable.tsx:
```diff
@@ -0,0 +1,47 @@
+import { loadEvaluations, updateEvaluations } from '../utils/csvHandler';
+
+// Implements User Story 2.2 by displaying the full list of evaluations in a sortable table.
+// Also supports User Story 1.2 – Edit and Delete
```
done

arunkjoyson commented on src/components/ConfirmationMessage.tsx:
```diff
@@ -0,0 +1,17 @@
+// Could support User Story 1.1 and 1.2 for showing feedback.
+import React from 'react';
+
+interface Props {
```
done

arunkjoyson commented on src/App.tsx:
done

arunkjoyson commented:
Since it's a big PR, decided to split it to multiple PR's and submit.
